<launch>
  <!-- Arguments -->
  <arg name="cmd_vel_topic" default="/cmd_vel" />
  <arg name="odom_topic" default="odom" />

  <!-- move_base -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">    
    <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />
    <rosparam file="$(find comp4034)/src/assignment/param/teb_local_planner_params_waffle.yaml" command="load" />
    <rosparam file="$(find comp4034)/src/assignment/param/costmap_common_params_waffle.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find comp4034)/src/assignment/param/costmap_common_params_waffle.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find comp4034)/src/assignment/param/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find comp4034)/src/assignment/param/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find comp4034)/src/assignment/param/move_base_params.yaml" command="load" />
    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>
  </node>
</launch>
